pyccel/ast/cwrapper.py

```diff
@@ -961,63 +953,65 @@ def declarations(self):
 
 # https://docs.python.org/3/c-api/list.html#c.PyList_GetItem
 PyList_GetItem = FunctionDef(name = 'PyList_GetItem',
                         body = [],
-                        arguments = [FunctionDefArgument(Variable(dtype=PyccelPyObject(), name='l', memory_handling='alias')),
-                                     FunctionDefArgument(Variable(dtype=NativeInteger(), name='i', precision=-2))],
-                        results = [FunctionDefResult(Variable(dtype=PyccelPyObject(), name='o', memory_handling='alias'))])
+                        arguments = [FunctionDefArgument(Variable(PyccelPyObject(), name='l', memory_handling='alias')),
+                                     FunctionDefArgument(Variable(CNativeInt(), name='i'))],
+                        results = [FunctionDefResult(Variable(PyccelPyObject(), name='o', memory_handling='alias'))])
 
 # https://docs.python.org/3/c-api/list.html#c.PyList_SetItem
 PyList_SetItem = FunctionDef(name = 'PyList_SetItem',
                         body = [],
-                        arguments = [FunctionDefArgument(Variable(dtype=PyccelPyObject(), name='l', memory_handling='alias')),
-                                     FunctionDefArgument(Variable(dtype=NativeInteger(), name='i', precision=-2)),
-                                     FunctionDefArgument(Variable(dtype=PyccelPyObject(), name='new_item', memory_handling='alias'))],
+                        arguments = [FunctionDefArgument(Variable(PyccelPyObject(), name='l', memory_handling='alias')),
+                                     FunctionDefArgument(Variable(CNativeInt(), name='i')),
+                                     FunctionDefArgument(Variable(PyccelPyObject(), name='new_item', memory_handling='alias'))],
                         results = [])
 
 #-------------------------------------------------------------------
 
 #using the documentation of PyArg_ParseTuple() and Py_BuildValue https://docs.python.org/3/c-api/arg.html
 pytype_parse_registry = {
-    (NativeInteger(), 4)       : 'i',
-    (NativeInteger(), 8)       : 'l',
-    (NativeInteger(), 2)       : 'h',
-    (NativeInteger(), 1)       : 'b',
-    (NativeFloat(), 8)         : 'd',
-    (NativeFloat(), 4)         : 'f',
-    (NativeComplex(), 4)       : 'O',
-    (NativeComplex(), 8)       : 'O',
-    (NativeBool(), -1)         : 'p',
-    (NativeString(), 0)        : 's',
-    (PyccelPyObject(), 0)      : 'O',
+    #(PythonNativeInt(), 4)       : 'i',
+    #(PythonNativeInt(), 8)       : 'l',
+    #(PythonNativeInt(), 2)       : 'h',
+    #(PythonNativeInt(), 1)       : 'b',
+    PythonNativeFloat() : 'd',
+    #(PythonNativeFloat(), 8)         : 'd',
+    #(PythonNativeFloat(), 4)         : 'f',
+    PythonNativeComplex() : 'O',
+    #(NativeComplex(), 4)       : 'O',
+    #(NativeComplex(), 8)       : 'O',
+    PythonNativeBool()         : 'p',
+    StringType()        : 's',
+    PyccelPyObject()      : 'O',
     }
```

Comment:
Should we remove the lines which you have commented out?

Reply:
These were commented until I move them to `numpy_wrapper.py`. I expected there to be an error if these values were not defined. I have moved them now: 2f8769932 but this probably means that this is either dead code or that we are lacking coverage. I have created #1788 to investigate this